### PR TITLE
use `polars` to compute stats over all datasets

### PR DESCRIPTION
**Pull request by polinaeterna**
\+ added unit-tests for each feature type
\+ added stats for boolean type (TODO: document + e2e test)

### CODE REVIEW

File: services/worker/src/worker/job_runners/split/descriptive_statistics.py [hunk 1]
```diff
@@ -148,110 +126,68 @@ def generate_bins(
...
-        data_table_name=table_name, bins_table_name=BINS_TABLE_NAME, column_name=column_name
-    )
-    logging.debug(f"Compute histogram for {column_name=}.\n{compute_hist_command}")
-    # query returns list of tuples (bin_id, bin_max, n_count):
-    hist_query_result = dict(con.sql(compute_hist_command).fetchall())  # dict bin_id -> n_samples
-    if len(hist_query_result) > n_bins + 1:
-        raise StatisticsComputationError(
-            f"Got unexpected result during histogram computation for {column_name=}: returned more bins than"
-            f" requested. {n_bins=} {hist_query_result=}. "
+    if len(bin_edges) == 2:  # possible if min == max (=data has only one value)
```
Comment (polinaeterna) [inline services/worker/src/worker/job_runners/split/descriptive_statistics.py:145]: i know that this is quite a corner case but i imagine it's possible

**Comment by lhoestq** [comment]
There was this minor change https://github.com/huggingface/datasets-server/pull/2189 recently, maybe we should check if it had any impact on blocking jobs for stats using duckdb before switching to polars

**Comment by polinaeterna** [comment]
@lhoestq 

> if it had any impact on blocking jobs for stats using duckdb before switching to polars

no, i think it didn't :( 
I rerun `c4`-`en`-`train` yesterday and got `{"error": "Job manager crashed while running this job (missing heartbeats)."}`

File: services/worker/src/worker/job_runners/split/descriptive_statistics.py [hunk 2]
```diff
@@ -148,110 +126,68 @@ def generate_bins(
...
+        if bin_edges[0] != bin_edges[1]:
+            raise StatisticsComputationError(
+                f"Got unexpected result during histogram computation for {column_name=}, {column_type=}: "
+                f" len({bin_edges=}) is 2 but {bin_edges[0]=} != {bin_edges[1]=}. "
+            )
+        hist = [int(df[column_name].is_not_null().sum())]
+    elif len(bin_edges) > 2:
+        bins_edges_reverted = [-1 * b for b in bin_edges[::-1]]
+        hist_df_reverted = df.with_columns(pl.col(column_name).mul(-1).alias("reverse"))["reverse"].hist(
```
[the remark this answers is not in the document]
Comment (severo) [reply services/worker/src/worker/job_runners/split/descriptive_statistics.py:161]: thanks!

it deserves a comment in the code

File: services/worker/src/worker/job_runners/split/descriptive_statistics.py [hunk 3]
```diff
@@ -646,7 +506,16 @@ def compute_descriptive_statistics_response(
     local_parquet_glob_path = Path(local_parquet_directory) / config / f"{split_directory}/*.parquet"
 
     num_examples = dataset_info["splits"][split]["num_examples"]
-    categorical_features = {
+    num_rows = pl.read_parquet(
+        local_parquet_glob_path, columns=[pl.scan_parquet(local_parquet_glob_path).columns[0]]
+    ).shape[0]
+    if num_rows != num_examples:
```
[the remark this answers is not in the document]
Comment (severo) [reply services/worker/src/worker/job_runners/split/descriptive_statistics.py:556]: I don't know. cc @lhoestq

**Review by severo — APPROVED** [review]

File: services/worker/src/worker/job_runners/split/descriptive_statistics.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (lhoestq) [reply services/worker/src/worker/job_runners/split/descriptive_statistics.py:556]: Yes it's the number of examples in the converted parquet files, so the `StatisticsComputationError` should not happen

**Review by lhoestq — APPROVED** [review]
Looks good to me ! Feel free to merge when you're good :)

File: docs/source/statistics.mdx [hunk 4]
```diff
@@ -304,6 +305,35 @@ The following measures are returned for integer data types:
...
+{
+  "column_name": "penalty",
+  "column_type": "bool",
+  "column_statistics":
+    {
+        "nan_count": 3,
+        "nan_proportion": 0.15,
+        "frequencies": {
+            'False': 7,
+            'True': 10
```
Comment (severo) [inline docs/source/statistics.mdx:328]: nit

JSON requires double quotes

Comment (polinaeterna) [reply docs/source/statistics.mdx:328]: ah thanks!

**Pull request merged**